dev/build-ballista-docker.sh

```diff
@@ -19,9 +19,13 @@
 
 set -e
 
+docker build -t ballista-builder --build-arg EXT_UID="$(id -u)" -f dev/docker/ballista-builder.Dockerfile .
+
+docker run -v $(pwd):/home/builder/workspace ballista-builder
```

Comment:
Build in the build container, but don't use ephemeral storage and wipe everything so we have faster build & test cycles, but CI is still pure.